### ISSUE TITLE
Ensure we're monitoring backend performance data for new Subscriptions service

### ISSUE THREAD

**Issue by clouserw** [mozilla/fxa#997]
**2019-09-30 EDIT (@chenba)** We've settled on using `statsd` (with the `hot-shots` npm module) to send stats and the InfluxDB stack to collect and query our data.

The following PRs add `statsd` timing calls and relevant configs:

- [x] https://github.com/mozilla/fxa/pull/2594
- [x] https://github.com/mozilla/fxa/pull/2631
- [x] https://github.com/mozilla/fxa/pull/2616
- [x] https://github.com/mozilla/fxa/pull/2652

TODOs before we can start sending stats from FxA:

- [ ] [Deploy Telegraf to collect metrics](https://github.com/mozilla/fxa/issues/2665)
- [x] [A BMO bug for Ops](https://bugzilla.mozilla.org/enter_bug.cgi?assigned_to=bpitts%40mozilla.com&bug_ignored=0&bug_severity=normal&bug_status=NEW&cf_fx_iteration=---&cf_fx_points=---&cf_status_firefox65=---&cf_status_firefox66=---&cf_status_firefox67=---&cf_status_firefox_esr60=---&cf_tracking_firefox65=---&cf_tracking_firefox66=---&cf_tracking_firefox67=---&cf_tracking_firefox_esr60=---&cf_tracking_firefox_relnote=---&comment=Introduction%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A%0D%0AThis%20is%20a%20tweaked%20version%20of%20the%20sizing%20questionnaire%20used%20by%20InfluxData.%20Filling%20it%20out%20is%20one%20of%20the%20first%20steps%20a%20new%20team%20needs%20to%20do%20to%20get%20onboarded%20to%20InfluxDB%20at%20Mozilla.%0D%0A%0D%0APlease%20answer%20the%20questions%20below%20to%20the%20best%20of%20your%20ability.%20%20We%20do%20not%20need%20exact%20answers%2C%20just%20estimates%20so%20we%20can%20get%20an%20idea%20of%20what%20your%20requirements%20are%20for%20sizing%20and%20architecture.%0D%0A%0D%0AIn%20order%20to%20answer%20these%20questions%2C%20it%20might%20be%20helpful%20to%20introduce%20some%20terminology%20to%20help%20ensure%20we%20are%20speaking%20the%20same%20language.%20%20In%20InfluxDB%20data%20is%20ingested%20and%20stored%20in%20what%20we%20call%20%E2%80%9Cline%20protocol%E2%80%9D.%20%20Line%20protocol%2C%20or%20an%20individual%20line%20is%20made%20up%20of%20a%20measurement%20name%2C%20a%20tag%20set%2C%20a%20field%20set%2C%20and%20a%20timestamp.%20%20Here%20is%20an%20example%20line%20of%20line%20protocol%3A%0D%0A%0D%0Acpu%2Ccpu%3Dcpu0%2Chost%3DtelegrafHost%20usage_user%3D2.59%2Cusage_system%3D4.59%2Cusage_idle%3D92.81%201509558725000000000%0D%0A%0D%0AThe%20measurement%20acts%20as%20a%20container%20for%20tags%2C%20fields%2C%20and%20the%20time%20column.%20In%20the%20example%20above%20our%20measurement%20name%20is%20cpu.%0D%0A%0D%0AThe%20tag%20set%20is%20the%20different%20combinations%20of%20all%20the%20tag%20key-value%20pairs.%20Tags%20are%20made%20up%20of%20tag%20keys%20and%20tag%20values%20and%20are%20indexed.%20%20Tags%20are%20metadata%20about%20our%20fields.%20%20In%20the%20example%20above%20our%20tagset%20is%3A%20cpu%3Dcpu0%2Chost%3DtelegrafHost%0D%0A%0D%0AThe%20field%20set%20is%20the%20different%20combinations%20of%20all%20the%20field%20key-value%20pairs.%20fields%20are%20made%20up%20of%20field%20keys%20and%20field%20values%20and%20are%20not%20indexed.%20%20Fields%20are%20what%20we%20are%20measuring.%20%20In%20the%20example%20above%20our%20fieldset%20is%3A%20usage_user%3D2.59%2Cusage_system%3D4.59%2Cusage_idle%3D92.81%0D%0A%0D%0AThe%20timestamp%20is%20the%20time%20associated%20with%20this%20line%20down%20to%20nanosecond%20precision.%20In%20the%20example%20above%20our%20timestamp%20is%3A%201509558725000000000%0D%0A%0D%0AA%20series%20is%20best%20thought%20of%20as%20a%20fingerprint%20or%20key%2C%20as%20an%20analogy.%20%20The%20series%20is%20defined%20by%20the%20measurement%20name%2C%20the%20tag%20set%2C%20and%20the%20field%20keys%0D%0A%0D%0AWhen%20we%20talk%20about%20ingest%20rates%20we%20will%20refer%20to%20values%20per%20second%20or%20values%20per%20minute.%20%20A%20value%20is%20an%20individual%20field%20key%2Fvalue%20pair.%20%20This%20is%20our%20atomic%20unit.%20%20We%20do%20not%20refer%20to%20lines%20per%20second%20or%20lines%20per%20minute%20because%20a%20line%20is%20made%20up%20of%20one%20or%20more%20values.%20%20Your%20line%20may%20vary%20in%20the%20number%20of%20values%20from%20another%20line.%0D%0A%0D%0AA%20datasource%20refers%20to%20an%20individual%20source%20that%20will%20be%20sending%20data%20to%20InfluxDB.%20%20This%20could%20be%20a%20host%20running%20Telegraf%2C%20or%20another%20agent%20pushing%20data.%0D%0A%0D%0A%0D%0A%0D%0AGeneral%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A1.%20What%20are%20the%20Mozilla%20email%20addresses%20of%20the%20people%20who%20need%20to%20manage%20dashboards%20and%20alerts%20for%20your%20team%3F%0D%0A%0D%0A1.%20What%20is%20a%20brief%20description%20of%20your%20use%20case%2C%20or%20how%20are%20you%20planning%20to%20use%20InfluxDB%3F%0D%0A%0D%0A1.%20When%20do%20you%20want%20to%20start%20on%20your%20implementation%3F%0D%0A%0D%0AWrites%0D%0A1.%20How%20many%20Data%20Sources%20do%20you%20have%20and%20what%20types%20are%20they%3F%20If%20you%20are%20monitoring%20hosts%20then%20how%20many%20hosts%20will%20you%20be%20monitoring%3F%0D%0A%0D%0A1.%20What%20is%20your%20proposed%20architecture%20for%20ingesting%20data%3F%20E.G.%20where%20will%20you%20run%20telegraf%2C%20what%20forwards%20to%20telegraf%20vs%20writing%20directly%20to%20influxdb%2C%20etc.%0D%0A%0D%0A1.%20How%20many%20values%20per%20second%20or%20minute%20will%20you%20be%20ingesting%20into%20the%20DB%3F%0D%0A%0D%0A1.%20If%20you%20are%20generating%20the%20data%2C%20what%20does%20your%20data%20look%20like%20%28please%20enter%20a%20sample%20of%20the%20data%20if%20possible%29%3F%0D%0A%0D%0A1.%20How%20often%20will%20you%20be%20sending%20data%20to%20the%20database%3F%20%20Every%20few%20seconds%2C%20minutes%2C%20etc.%0D%0A%0D%0A1.%20How%20many%20unique%20series%20do%20you%20think%20you%20will%20have%3F%0D%0A%0D%0A%0D%0AReads%0D%0A%3D%3D%3D%3D%3D%3D%0D%0A1.%20What%20sort%20of%20query%20volume%20do%20you%20anticipate%20having%20%28queries%2Fsec%20or%20min%29%3F%0D%0A%0D%0A1.%20How%20many%20concurrent%20queries%20do%20you%20expect%20to%20have%3F%0D%0A%0D%0A1.%20What%20types%20of%20queries%20do%20you%20expect%20to%20run%20%28dashboards%20or%20analytics%29%3F%0D%0A%0D%0A1.%20How%20many%20users%20will%20be%20using%20the%20dashboards%20at%20any%20point%20in%20time%3F%0D%0A%0D%0A1.%20Do%20you%20have%20any%20requirements%20for%20response%20times%20on%20these%20queries%3F%0D%0A%0D%0A1.%20What%20sort%20of%20other%20analytic%20or%20downsampling%20jobs%20do%20you%20expect%20to%20have%3F%0D%0A%0D%0ARetention%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A%0D%0A1.%20How%20long%20do%20you%20need%20to%20retain%20the%20data%20at%20full%20resolution%3F%20For%20example%2C%2015%20days%0D%0A%0D%0A1.%20Do%20you%20need%20to%20keep%20downsampled%20data%20for%20longer%3F%20If%20so%2C%20on%20what%20schedule%3F%20For%20example%2C%205%20minute%20averages%20for%2063%20days%2C%201%20hour%20averages%20for%20366%20days%0D%0A%0D%0A%0D%0AProcessing%2C%20analytics%2C%20alerting%20and%20taking%20action%0D%0A%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%3D%0D%0A1.%20What%20kind%20of%20processing%2C%20if%20any%2C%20is%20required%20on%20incoming%20data%3F%0D%0A%0D%0A1.%20Do%20you%20do%20any%20alerting%20based%20on%20patterns%2C%20anomalies%20or%20thresholds%20crossed%3F&component=Operations%3A%20Metrics%2FMonitoring&contenttypemethod=list&contenttypeselection=text%2Fplain&defined_groups=1&flag_type-37=X&flag_type-5=X&flag_type-607=X&flag_type-708=X&flag_type-721=X&flag_type-737=X&flag_type-748=X&flag_type-787=X&flag_type-800=X&flag_type-803=X&flag_type-846=X&flag_type-864=X&flag_type-929=X&flag_type-935=X&flag_type-936=X&form_name=enter_bug&maketemplate=Remember%20values%20as%20bookmarkable%20template&op_sys=Unspecified&priority=--&product=Cloud%20Services&rep_platform=Unspecified&short_desc=InfluxDB%20Onboarding%20for%20YOURTEAMHERE&target_milestone=---&version=unspecified)

**Original Description** 
We can use re:dash for this one too, but let's make sure we're recording performance data.  I'm thinking of round-trip times for talking to SubHub APIs and how many items are in the queues, for starters.  There are probably other things we should be watching too.

**Comment by chenba** [comment]
While there are precedents of using flow events for [performance metrics](https://sql.telemetry.mozilla.org/dashboard/fxa-performance), they are especially not a good fit for measuring backend service API call performance.  After receiving [some comments from Phil](https://github.com/mozilla/fxa/pull/2474#discussion_r323390192) on my draft PR #2474 (and some follow-up chat on Slack), [the idea of a brand new event type](https://github.com/mozilla/fxa/pull/2474#discussion_r323427688) is more appealing than ever.

**Comment by bbangert** [comment]
> Looks great, but IIUC it only measures back-end performance? I think there's value in measuring network performance too, because that impacts users and we want to know about it if it deteriorates.

Yup, this issue is for backend performance though. We have a separate issue for front-end perf.

**Comment by chenba** [comment]
Unfortunately I was not able to find a way to view the performance of http/redis/whatever calls by themselves in Stackdriver Trace.  The subspans can be used to filter for requests (e.g. all requests that has a call to SubHub's plans endpoint), but I don't see a way of viewing them in isolation.

**Comment by chenba** [comment]
I am going to close this issue since the code's in place now.  I will file an issue for creating charts in Chronograf as a follow-up to Telegraf.
